interface/src/app/gene/searchMany/searchMany.js

```diff
@@ -264,94 +189,80 @@ angular.module('adage.gene.searchMany', [
 .directive('geneResultButton', [function() {
   return {
     controller: [
-      '$scope', 'SearchResults', 'SelectedGenesFactory', 'CommonGeneFuncts',
-      function($scope, SearchResults, SelectedGenesFactory, CommonGeneFuncts) {
+      '$scope', 'SelectedGenesFactory', 'CommonGeneFuncts',
+      function($scope, SelectedGenesFactory, CommonGeneFuncts) {
         $scope.geneLabel = CommonGeneFuncts.getGeneLabel($scope.gene);
 
         $scope.addToSelectedGenes = function() {
           SelectedGenesFactory.addGene($scope.gene);
-          SearchResults.remove($scope.query);
-        };
-      }
-    ],
-    restrict: 'E',
-    replace: true,
-    templateUrl: 'gene/searchMany/gene-result-button.tpl.html'
-  };
-}])
-
-// Directive for button to get more options, should get
-// next page of search results for this query from the server
-.directive('moreResultButton', [function() {
-  return {
-    controller: ['$scope', 'SearchResults', 'CommonGeneFuncts',
-      function($scope, SearchResults, CommonGeneFuncts) {
-        $scope.nextGenePage = function() {
-          $scope.page = $scope.pageDict.page + 1;
-          CommonGeneFuncts.updatePageNumbers($scope);
-        };
-      }
-    ],
-    replace: true,
-    restrict: 'E',
-    scope: false,
-    templateUrl: 'gene/searchMany/more-result-button.tpl.html'
-  };
-}])
-
-// Directive for button to get previous search results
-.directive('previousResultButton', [function() {
-  return {
-    controller: ['$scope', 'SearchResults', 'CommonGeneFuncts',
-      function($scope, SearchResults, CommonGeneFuncts) {
-        $scope.previousGenePage = function() {
-          $scope.page = $scope.pageDict.page - 1;
-          CommonGeneFuncts.updatePageNumbers($scope);
+          delete $scope.searchResults[$scope.query];
         };
       }
     ],
+    scope: {
+      query: '@',
+      gene: '=',
+      searchResults: '=',
+      queries: '='
+    },
     replace: true,
```

Comment:
Not really related to this PR: The keyword **replace** has been deprecated since Angular 1.3. You may want to avoid it.